backend/coreapp/compiler_wrapper.py

```diff
@@ -34,14 +29,13 @@ def run_compiler(compile_cmd: str, input_path: Path, output_path: Path, compiler
             .replace("$OUTPUT", str(output_path)) \
             .replace("$COMPILER_PATH", str(compiler_path)) \
             .replace("$CC_FLAGS", cc_flags)
-        compile_command = f"bash -c \"{compile_command}\""
 
         logger.debug(f"Compiling: {compile_command}")
 
         return_code = 0
 
         try:
-            result = subprocess.run(compile_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+            result = subprocess.run(["bash", "-c", compile_command], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
This removes a shell layer (`shell=True` is basically `bash -c ...`), and helps avoid some escaping issues in `compile_command`.

Reply:
great!